src/nvim/ops.c

```diff
@@ -715,150 +703,6 @@ static void block_insert(oparg_T *oap, const char *s, size_t slen, bool b_insert
-  nested--;
-  xfree(expr_copy);
-  return rv;
-}
-
-/// Get the '=' register expression itself, without evaluating it.
-char *get_expr_line_src(void)
-{
-  if (expr_line == NULL) {
-    return NULL;
-  }
-  return xstrdup(expr_line);
-}
-
-/// @return  whether `regname` is a valid name of a yank register.
-///
-/// @note: There is no check for 0 (default register), caller should do this.
-/// The black hole register '_' is regarded as valid.
-///
-/// @param regname name of register
-/// @param writing allow only writable registers
-bool valid_yank_reg(int regname, bool writing)
-{
-  if ((regname > 0 && ASCII_ISALNUM(regname))
-      || (!writing && vim_strchr("/.%:=", regname) != NULL)
-      || regname == '#'
-      || regname == '"'
-      || regname == '-'
-      || regname == '_'
-      || regname == '*'
-      || regname == '+') {
-    return true;
-  }
-  return false;
-}
-
-/// @return yankreg_T to use, according to the value of `regname`.
-/// Cannot handle the '_' (black hole) register.
-/// Must only be called with a valid register name!
-///
-/// @param regname The name of the register used or 0 for the unnamed register
-/// @param mode One of the following three flags:
-///
-/// `YREG_PASTE`:
-/// Prepare for pasting the register `regname`. With no regname specified,
-/// read from last written register, or from unnamed clipboard (depending on the
-/// `clipboard=unnamed` option). Queries the clipboard provider if necessary.
-///
-/// `YREG_YANK`:
-/// Preparare for yanking into `regname`. With no regname specified,
-/// yank into `"0` register. Update `y_previous` for next unnamed paste.
-///
-/// `YREG_PUT`:
-/// Obtain the location that would be read when pasting `regname`.
-yankreg_T *get_yank_register(int regname, int mode)
-{
-  yankreg_T *reg;
-
-  if ((mode == YREG_PASTE || mode == YREG_PUT)
-      && get_clipboard(regname, &reg, false)) {
-    // reg is set to clipboard contents.
-    return reg;
-  } else if (mode == YREG_PUT && (regname == '*' || regname == '+')) {
-    // in case clipboard not available and we aren't actually pasting,
-    // return an empty register
-    static yankreg_T empty_reg = { .y_array = NULL };
-    return &empty_reg;
-  } else if (mode != YREG_YANK
-             && (regname == 0 || regname == '"' || regname == '*' || regname == '+')
-             && y_previous != NULL) {
-    // in case clipboard not available, paste from previous used register
-    return y_previous;
-  }
-
-  int i = op_reg_index(regname);
-  // when not 0-9, a-z, A-Z or '-'/'+'/'*': use register 0
-  if (i == -1) {
-    i = 0;
-  }
-  reg = &y_regs[i];
-
-  if (mode == YREG_YANK) {
-    // remember the written register for unnamed paste
-    y_previous = reg;
-  }
-  return reg;
-}
-
-static bool is_append_register(int regname)
-{
-  return ASCII_ISUPPER(regname);
-}
-
 /// @return  a copy of contents in register `name` for use in do_put. Should be
 ///          freed by caller.
 yankreg_T *copy_register(int name)
```

Comment:
`copy_register()` was renamed from `get_register()` in edb13791bd6eda43ab3816c7697530b378680e61. `get_register()` was moved in this patch, so `copy_register()` should be moved to register.c.